  function testcase() 
  {
    var called = 0;
    function callbackfn(val, idx, obj) 
    {
      called++;
      if (called === 1)
      {
        throw new Error("Exception occurred in callbackfn");
      }
      return true;
    }
    var obj = {
      0 : 11,
      4 : 10,
      10 : 8,
      length : 20
    };
    try
{      
// TODO: add CFGEnterCode to javascript modeling	
	Array.prototype.filter.call(obj, callbackfn);
    return false;
}
    catch (ex)
{      return 1 === called && ex instanceof Error;}

  }
  {
    var __result1 = testcase();
    var __expect1 = true;
  }
  
